src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i !== 0 && i < this.length) {
+        string += separator;
+      }
```

Comment:
You can add a separator after the element. It needs only to check `i` is less than `length - 1`.